sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/WorkerBase.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.export;
+
+import io.opentelemetry.api.metrics.BoundLongCounter;
+import io.opentelemetry.api.metrics.GlobalMetricsProvider;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public abstract class WorkerBase implements Runnable {
```

Comment:
I don't think I'm comfortable making this a part of our public SDK APIs at this point.

Reply:
@jkwatson I will try to put this class in package scope. Would that be OK?

Reply:
I'm not sure that will help, since the Java module system doesn't let us have the same package in two jars loaded into the same classloader.

Reply:
Ok, so I will remove this class. There will be some code duplication but at least we will avoid other problems.

Reply:
If we put this class into an incubator module, and all the BSP options are in that module, in the same package, then that would work.

Reply:
@jkwatson do you mean `sdk-extensions/tracing-incubator` gradle module?

Reply:
yes, that seems like the right place to incubate sdk bits, like optional span processors.

Reply:
Ok, I've moved the code to the incubator and reverted changes I've made to the original `BatchSpanProcessor`